### ISSUE TITLE
v2.1.0-pre1 - Graphs don't render on empty InfluxDB resultset

### ISSUE THREAD

**Issue by Inisfair** [grafana/grafana#2294]
When creating dashboards that include a valid query that returns an empty result set from InfluxDB, a javascript error occurs:
```
Message:
			Cannot read property 'length' of undefined
		
Stack trace:
			TypeError: Cannot read property 'length' of undefined
    at b.c.getTimeSeries (http://dashboard01.example.com/public/app/plugins/datasource/influxdb/influxSeries.js?bust=1436199875737:4:214)
    at http://dashboard01.example.com/public/app/plugins/datasource/influxdb/datasource.js?bust=1436199875737:4:1652
    at i (http://dashboard01.example.com/app/app.da210931.js:12:3146)
    at http://dashboard01.example.com/app/app.da210931.js:12:3318
    at n.$eval (http://dashboard01.example.com/app/app.da210931.js:12:10922)
    at n.$digest (http://dashboard01.example.com/app/app.da210931.js:12:9353)
    at n.$apply (http://dashboard01.example.com/app/app.da210931.js:12:11189)
    at f (http://dashboard01.example.com/app/app.da210931.js:11:17908)
    at r (http://dashboard01.example.com/app/app.da210931.js:11:19705)
    at XMLHttpRequest.v.onload (http://dashboard01.example.com/app/app.da210931.js:11:20245)
```

The InfluxDB query is valid but when it contains no results it will cause the entire panel to fail to render instead of simply not plotting the empty resultset.

```
inisfair@dashboard01 => curl -G 'http://influxdb01:8086/query' --data-urlencode "db=graphite_raw" --data-urlencode "q=SELECT sum(value) FROM transaction_count WHERE app = 'app1' AND role =~ /function1|all/ AND environment = 'prod' AND datacenter = 'detroit' AND time > now() - 6h GROUP BY time(5m), status fill(null) ORDER BY asc"
{"results":[{}]}
[~]
```

as opposed to an invalid query which returns an error from influxdb:
```
inisfair@dashboard01 => curl -G 'http://influxdb01:8086/query' --data-urlencode "db=graphite_raw" --data-urlencode "q=SELECT sum(value) FROM transaction_count WHERE app2 = 'app1' AND role =~ /function1|all/ AND environment = 'prod' AND datacenter = 'detroit' AND time > now() - 6h GROUP BY time(5m), status fill(null) ORDER BY asc"
{"results":[{"error":"unknown field or tag name in where clause: app2"}]}
[~]
```

In both query scenarios, the same javascript error causes the panel to fail to render at all.

**Comment by Inisfair** [comment]
I should note that this only occurs on a panel with multiple metrics, on a panel with only the empty resultset metric you get the expected "no datapoints" indication

**Comment by torkelo** [comment]
what build are you on? I cannot replicate this with latest master build

**Comment by Inisfair** [comment]
Installed from the 20150630 RPM of v2.1.0-pre1.  If there is a more specific build info I can get somewhere let me know?

I'm installing the v2.1.0-pre1 RPM dated today to see if it addresses the issue and will report back.

**Comment by Inisfair** [comment]
Same issue after upgrading to 20150706 build date RPM

**Comment by Inisfair** [comment]
[Screencast](http://screencast.com/t/vipCBvcSb9c)

**Comment by torkelo** [comment]
@Inisfair can you show the response from InfluxDB when you get the error? I have tried with a complete empty result set, with multiple queries where one result set is emtpy. Both scenarios work fine. So it must be something strange with the response from InfluxDB

**Comment by Inisfair** [comment]
In the pop-up inspector both the request and response panels are blank.

In my environment I have a graph panel that has 4 series.  The first 3 are the avg/max/80th percentile of CPU utilization and the 4th is transaction volume grouped by status (so can be 1 or more actual chart series).

I set the datasource to direct so I can use Wireshark to capture the conversation between my browser and InfluxDB, in the reply from InfluxDB I can see the result set for the first 3 series, the result set for the 4th query is an empty result set.

The failing JSON looks like this in structure:
```
{
  "results": [
    {
      "series": [...]
    },
    {
      "series": [...]
    },
    {
      "series": [...]
    },
    {}
  ]
}
```
If I remove the first 3 series and have just the transaction_count query I get this as the (full) response:
```
{"results":[{}]}
```

**Comment by Inisfair** [comment]
Headers/Response from chrome developer tols...

Good: https://gist.github.com/Inisfair/1494d2c54d598fd8d780
Bad: https://gist.github.com/Inisfair/e8c9baca18fd51f45d45

**Comment by timgriffiths** [comment]
I seen this as well, it happens when you don't have a measurement (or series in the old influx terminology) in the InfluxDB, ie if you do a SHOW MEASUREMENTS against your influx database i would bet "transaction_count" does not exist

I agree it's a bit unfortunate that grafana will not display the other valid results

**Comment by Inisfair** [comment]
transaction_count does exist in the "show measurments" output however the specific combination of app/role tag does not exist in "show series" or "show tag values from transaction_count where app = 'appname' and role = 'approle'" output.  Running the query via returns an empty result set.  You can also see the empty result set at the end of the json in the "bad" output.

The intent is to have a generic series to depict the workload (e.g. the transactions) against the utilization (cpu) automatically when the transaction data is available without having to edit the charts, as over time we are adding more and more app/role entries to the transaction_count measurement.

**Comment by ecables** [comment]
Same problem here.

**Comment by jdunjo** [comment]
Hi, we've the same error when trying to set a TAG value that it doesn't exist yet. In the pop-up inspector both the request and response panels are blank and we only can read the following error: 

<addr>
Message:
			Cannot read property 'length' of undefined
		
Stack trace:
			TypeError: Cannot read property 'length' of undefined
    at b.c.getTimeSeries (http://grafana.xmltravelgate.com:3000/public/app/plugins/datasource/influxdb/influxSeries.js?bust=1438244173402:4:218)
</addr>

Basically is the same problem that Inisfair reported (using InfluxDB 0.9.x datasource). Do you know if some future release will fix this issue? We just need to show the right metrics that return results.

**Comment by torkelo** [comment]
manged to replicate this now, not sure why I could not replicate it before. Anyway its fixed and new builds are being build and should appear shortly
